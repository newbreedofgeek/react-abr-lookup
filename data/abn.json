{
   "Abn":"74172177893",
   "AbnStatus":"Active",
   "AddressDate":"2011-06-23",
   "AddressPostcode":"2601",
   "AddressState":"ACT",
   "BusinessName":[

   ],
   "EntityName":"THE TRUSTEE FOR PSS FUND",
   "EntityTypeCode":"CSP",
   "EntityTypeName":"Commonwealth Government APRA Regulated Public Sector Scheme",
   "Gst":"2000-07-01",
   "Message":""
}