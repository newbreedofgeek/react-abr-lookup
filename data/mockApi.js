export const abn = 'callback({"Abn":"26008672179","AbnStatus":"Active","AddressDate":"2018-02-26","AddressPostcode":"3123","AddressState":"VIC","BusinessName":["Hibberson Street Centre","home@gladesville","Gladesville Homemaker Centre","Bunnings Outdoors","big prawn","G.Y.O.","BUNNINGS TRADE","BUNNINGS TRADE","BATEMANS BAY HOMEMAKER CENTRE","BUNNINGS","BALLINA HOME CENTRE","NARELLAN HOMEMAKER CENTRE","TAREE HOME CENTRE","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","BUNNINGS HOMEMAKER CENTRE","BUNNINGS SUPA CENTA","DUBBO HOMEMAKER CENTRE","BUNNINGS","BUNNINGS","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","BUNNINGS WAREHOUSE","HARDWARE CAFE","HARDWARE CAFE"],"EntityName":"BUNNINGS GROUP LIMITED","EntityTypeCode":"PUB","EntityTypeName":"Australian Public Company","Gst":"2000-07-01","Message":""})';
export const name = 'callback({"Message":"","Names":[{"Abn":"26008672179","AbnStatus":"0000000001","IsCurrent":true,"Name":"BUNNINGS","NameType":"Business Name","Postcode":"3123","Score":100,"State":"VIC"},{"Abn":"63099797977","AbnStatus":"0000000002","IsCurrent":true,"Name":"BUNNINGS HANDICRAFTS","NameType":"Trading Name","Postcode":"3105","Score":97,"State":"VIC"}]})';