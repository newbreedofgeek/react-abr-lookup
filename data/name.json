{
   "Message":"",
   "Names":[
      {
         "Abn":"74599608295",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY",
         "NameType":"Entity Name",
         "Postcode":"2601",
         "Score":100,
         "State":"ACT"
      },
      {
         "Abn":"72189919072",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY",
         "NameType":"Entity Name",
         "Postcode":"2800",
         "Score":100,
         "State":"NSW"
      },
      {
         "Abn":"84144814556",
         "AbnStatus":"0000000002",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRIAL RELATIONS",
         "NameType":"Trading Name",
         "Postcode":"2010",
         "Score":97,
         "State":"NSW"
      },
      {
         "Abn":"52293849579",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRIAL RELATIONS",
         "NameType":"Other Name",
         "Postcode":"4000",
         "Score":97,
         "State":"QLD"
      },
      {
         "Abn":"84085734992",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRIES AND BUSINESS",
         "NameType":"Other Name",
         "Postcode":"0810",
         "Score":97,
         "State":"NT"
      },
      {
         "Abn":"69410335356",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY AND RESOURCES",
         "NameType":"Entity Name",
         "Postcode":"6004",
         "Score":97,
         "State":"WA"
      },
      {
         "Abn":"74599608295",
         "AbnStatus":"0000000001",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY AND SCIENCE",
         "NameType":"Entity Name",
         "Postcode":"2601",
         "Score":97,
         "State":"ACT"
      },
      {
         "Abn":"94200796476",
         "AbnStatus":"0000000002",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY AND TECHNOLOGY",
         "NameType":"Entity Name",
         "Postcode":"6000",
         "Score":97,
         "State":"WA"
      },
      {
         "Abn":"64739088312",
         "AbnStatus":"0000000002",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF INDUSTRY AND TRADE",
         "NameType":"Trading Name",
         "Postcode":"5000",
         "Score":97,
         "State":"SA"
      },
      {
         "Abn":"78342684030",
         "AbnStatus":"0000000002",
         "IsCurrent":true,
         "Name":"DEPARTMENT OF PRIMARY INDUSTRIES",
         "NameType":"Trading Name",
         "Postcode":"4000",
         "Score":97,
         "State":"QLD"
      }
   ]
}