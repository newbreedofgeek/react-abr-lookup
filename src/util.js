export const standardizeResult = (results) => {
  const adjusted = Array.isArray(results) ? results : (results.Abn === '') ? [] : [results];

  return adjusted.map(i => {
    return {
      name: (i.EntityName) ? i.EntityName : i.Name,
      abn: i.Abn,
      abnStatus: i.AbnStatus
    }
  })
}