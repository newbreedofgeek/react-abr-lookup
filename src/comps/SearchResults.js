import React, { Fragment } from "react";
import { abnLookup, acnLookup, nameLookup } from "../common/api";
import { createResource } from "../common/cache";
import { Spinner } from './Spinner';
import { standardizeResult } from '../common/util';

const abnLookupResource = createResource(abnLookup);
const acnLookupResource = createResource(acnLookup);
const nameLookupResource = createResource(nameLookup);

export default ({ term, type, loadingId, onResultClick }) => {
  const results = standardizeResult(
    (type === 'abn') ? abnLookupResource.read(term) : (type === 'acn' ? acnLookupResource.read(term) : nameLookupResource.read(term))
  );

  return (
    <Fragment>
      {results.length === 0 ? <span>Nothing found...</span> 
        : results.map((result) => (
        <div
          className="entity"
          key={`${result.abn}-${Date.now()}`}
          onClick={() => onResultClick(result.abn)}
        >
          <div className="main">
            <div className="title">{result.name}</div>
            <div className="info">{`ABN: ${result.abn}`}</div>
          </div>
          {loadingId !== result.abn && <div className="hover">{'ðŸ‘‰'}</div>}
          {loadingId === result.abn && (
            <div className="loading">
              <Spinner size="small" />
            </div>
          )}
        </div>
      ))}
    </Fragment>
  );
};