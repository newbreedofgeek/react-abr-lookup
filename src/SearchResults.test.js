import React, { Suspense } from 'react';
import SearchResults from './SearchResults';
import renderer from 'react-test-renderer';
import { abn as mockAbnApi } from '../data/mockApi';
import { delay } from './util';

beforeEach(() => {
  fetch.resetMocks()
})

test('SearchResults renders as intended', async done => {
  let mockedRes = mockAbnApi;

  fetch.mockResponseOnce(mockedRes);

  const component = renderer.create(
    <Suspense maxDuration={1000} fallback={<span>loading</span>}>
      <SearchResults
        term={26008672179} 
        type={'abn'} 
        loadingId={26008672179} 
        onResultClick={() => (true)} />
    </Suspense>
  );

  // should be in loading
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // react-test-renderer does not seem to support concurrent Suspense
  // ... but as we mock all we need to do is pull the snapshot out of the current thread to work
  // ... so below should worlk with a delay of 1ms but keeping to 500
  await delay(500);

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  done();
});