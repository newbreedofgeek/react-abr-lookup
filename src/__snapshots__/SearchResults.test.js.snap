// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders as intended 1`] = `
<span>
  loading
</span>
`;

exports[`SearchResults renders as intended 2`] = `
<div
  className="entity"
  onClick={[Function]}
>
  <div
    className="main"
  >
    <div
      className="title"
    >
      BUNNINGS GROUP LIMITED
    </div>
    <div
      className="info"
    >
      ABN: 26008672179
    </div>
  </div>
  <div
    className="hover"
  >
    ðŸ‘‰
  </div>
</div>
`;
